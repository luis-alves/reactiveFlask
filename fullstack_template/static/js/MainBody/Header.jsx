import React from "react"
import AccountName from "./Header/Account/AccountName"


class Header extends React.Component {
  constructor(props) {
    super(props)
    // this.props.cleared_balance = {'total': 0.0}
    this.state = {cleared: 0, uncleared: 0, working: 0}
    this.getValor = this.getValor.bind(this)
  }

  getValor() {
    fetch('account',
        )
        .then(response => {
            if (response.status >= 400) {
                throw new Error("Bad response from server");
            }
            response.json().then(data => {
              let cleared = 0
              let uncleared = 0
              let working = 0
              data.map(item => {
                let inflow = parseFloat(item['inflow'])
                let outflow = parseFloat(item['outflow'])
                if (item['reconcile'] == 'true') {
                  cleared += inflow - outflow
                }
                else if (item['reconcile'] == 'false') {
                  uncleared += inflow - outflow
                }
                working = cleared + uncleared
              })
              this.setState({cleared: cleared.toFixed(2),
                             uncleared: uncleared.toFixed(2),
                             working: working.toFixed(2)
                           })
            })
        });
  }

  componentDidMount () {
    this.getValor()
  }

  render () {
    return (
      <div className="page-header">
        <AccountName balance={this.state}/>
      </div>

    )
  }
}

export default Header
