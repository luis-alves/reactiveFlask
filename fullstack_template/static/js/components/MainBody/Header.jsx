import React from "react"
import ClearedBalance from "./Header/Account/ClearedBalance"
import UnclearedBalance from "./Header/Account/UnclearedBalance"
import WorkingBalance from "./Header/Account/WorkingBalance"



class Header extends React.Component {
  constructor(props) {
    super(props)
  }

  render () {
    return (
      <div className="page-header">
        <div className="account-name-container">
          <ClearedBalance />
          <div className="balance-separator">
            +
          </div>
          <UnclearedBalance />
          <div className="balance-separator">
            =
          </div>
            <WorkingBalance />
          </div>


      </div>

    )
  }
}

export default Header
