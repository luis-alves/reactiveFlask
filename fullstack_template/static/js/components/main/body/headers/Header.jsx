import React from 'react'

import Balance from './account/Balance'


class Header extends React.Component {
  render () {
    return (
      <div className='page-header'>
        <Balance />
      </div>
    )
  }
}

export default Header
