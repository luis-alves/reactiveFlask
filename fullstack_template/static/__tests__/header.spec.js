import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Header from '../js/components/main/body/headers/Header.jsx';

describe('Header component', () => {
  it('renders without exploding', () => {
    expect(
      shallow(<Header />).length).toEqual(1)
  })
  // it('Header has class page-header', () => {
  //   const component = shallow(<Header />);
  //   expect(component.find('.page-header').length).toEqual(1);
  // })
})

test('Check snapshot', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});
