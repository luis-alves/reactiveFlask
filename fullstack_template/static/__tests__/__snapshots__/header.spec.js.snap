// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check snapshot 1`] = `
<div
  className="page-header"
>
  <Connect(Balance) />
</div>
`;
